backend/controllers/session.js

```diff
@@ -166,6 +166,60 @@ module.exports.loginStudent = async (req, res) => {
     });
 };
 
+module.exports.resetPassword = async (req, res) => {
+    if (!req.user) {
+        return res.status(403).json({
+            status: 'error',
+            message: 'You are not signed in.',
+        });
+    }
+    const teacher = await Teacher.findById(req.user._id);
+
+    if (!teacher) {
+        // We have a random delay to prevent time-attacks
+        setTimeout(() => {
+            return res.status(403).json({
+                status: 'error',
+                message: 'You are not signed in as a Teacher.',
+            });
+        }, Math.random() * 100);
```

Comment:
We don't really need this `setTimeout()` here. It's really only meant to obfuscate whether an email address exists or not on the login page. At this point, we know the user exists and is valid, so there's no point is hiding that their account exists...